substrate/frame/support/src/storage/mod.rs

```diff
@@ -167,6 +167,47 @@ pub trait StorageValue<T: FullCodec> {
 	{
 		T::decode_len(&Self::hashed_key())
 	}
+
+	/// Read the length of the storage value without decoding the entire value.
+	///
+	/// `T` is required to implement [`StorageDecodeNonDedupLength`].
+	///
+	/// If the value does not exists or it fails to decode the length, `None` is returned.
+	/// Otherwise `Some(len)` is returned.
+	///
+	/// # Warning
+	///
+	/// - The value returned is the non-deduplicated length of the underlying Vector in storage.
+	/// This means that any duplicate items are included.
+	///
+	/// - `None` does not mean that `get()` does not return a value. The default value is completely
+	/// ignored by this function.
+	/// # Example
+	///
+	/// ```text
```

Comment:
Why is this just text? Does it not compile?  
You can also embed a test to ensure that it always works: https://github.com/paritytech/polkadot-sdk/blob/9485b0b4921a0ed21ef6cf21cf1270aa91cfc94e/substrate/frame/bags-list/src/lib.rs#L122

Reply:
Yep, it didn't compile because `TestExternalities` wasn't found. I'll get it to compile and remove `text`